pallets/transaction-payment/src/lib.rs

```diff
@@ -258,6 +258,57 @@ where
     }
 }
 
+pub trait ChargeTxFee {
+    fn charge_fee(who: Signatory, len: u32, info: DispatchInfo) -> TransactionValidity;
+}
+
+impl<T: Trait> ChargeTxFee for Module<T> {
+    fn charge_fee(who: Signatory, len: u32, info: DispatchInfo) -> TransactionValidity {
+        let fee = if info.pays_fee {
```

Comment:
isn't it possible to use the `compute_fee()` logic here to avail the same consistency in the fee calculation?

Reply:
Not really, `compute_fee` requires `send` and `sync` traits that help in concurrency. Our execution instance is not meant to be executed concurrently and hence does not have these traits. Defining these manually is unsafe. An alternative is to make `compute_fee` (and as a result, the charge fee signed extension) single-threaded but that will cause a performance penalty. That's why I opted for this approach that requires a bit of code duplication but is secure and performant.